freqtrade/persistence/models.py

```diff
@@ -847,7 +847,10 @@ def calc_profit_ratio(self, rate: Optional[float] = None,
 
     def recalc_trade_from_orders(self):
         # We need at least 2 entry orders for averaging amounts and rates.
+        # TODO: this condition could probably be removed
         if len(self.select_filled_orders(self.enter_side)) < 2:
+            self.stake_amount = self.amount * self.open_rate / self.leverage
```

Comment:
I've moved this here for the moment.

While the "proper" approach would be to remove this condition and run the same code throughout - it's breaking too many tests at the moment due to "mock missmatch" (we mock a order that's slightly different to what we'd get here).

I have another idea on how to handle this - but i'll not add that to feat/short anymore, therefore we'll take this shortcut for the moment.